pkg/driver/trace_driver.go

```diff
@@ -571,19 +799,28 @@ func (d *Driver) internalRun(iatOnly bool, generated bool) {
 
 	if !iatOnly {
 		log.Info("Generating IAT and runtime specifications for all the functions")
+		maxInvocation := getMaxInvocation(d.Configuration.Functions)
 		for i, function := range d.Configuration.Functions {
-			// Equalising all the InvocationStats to the first function
-			if d.Configuration.LoaderConfiguration.DAGMode {
-				function.InvocationStats.Invocations = d.Configuration.Functions[0].InvocationStats.Invocations
-			}
 			spec := d.SpecificationGenerator.GenerateInvocationData(
 				function,
 				d.Configuration.IATDistribution,
 				d.Configuration.ShiftIAT,
 				d.Configuration.TraceGranularity,
 			)
-
 			d.Configuration.Functions[i].Specification = spec
+			// Overwriting the runtime specification to account for maximum possible invocations
+			if d.Configuration.LoaderConfiguration.DAGMode {
+				originalInvocation := function.InvocationStats.Invocations
+				function.InvocationStats.Invocations = maxInvocation
+				spec := d.SpecificationGenerator.GenerateInvocationData(
+					function,
+					d.Configuration.IATDistribution,
+					d.Configuration.ShiftIAT,
+					d.Configuration.TraceGranularity,
+				)
+				function.InvocationStats.Invocations = originalInvocation
+				function.Specification.RuntimeSpecification = spec.RuntimeSpecification
+			}
```

Comment:
Could you kindly explain the reason for overwriting the runtime specification to account for maximum possible invocations?

Reply:
Sure! From what I observe, ```GenerateInvocationData()``` generates a matrix ```spec.RuntimeSpecification``` consisting of the execution specs (RunTime and Memory). This matrix is accessed with the index [MinuteIndex][InvocationIndex]. In a DAG invocation, the frequency of Invocation is determined by the root function, which has a different invocation frequency than the other functions. Thus, to prevent ```error: index out of range``` , each function's matrix needs to account for the maximum possible invocations, which is the maximum frequency of all the functions.